src/deploy/NVA_build/noobaa_init.sh

```diff
@@ -151,16 +151,37 @@ prepare_server_pvs() {
 }
 
 prepare_mongo_pv() {
-  local dir="/mongo_data/mongo/cluster/shard1"
+  local main_dir="/mongo_data"
+  local shard_dir="/mongo_data/mongo/cluster/shard1"
+  local uid=$(stat -c '%g' ${main_dir})
+  local gid=$(stat -c '%u' ${main_dir})
 
   # change ownership and permissions of mongo db path
-  ${KUBE_PV_CHOWN} mongo
+  if [ "${uid}" != "10001" ] || [ "${gid}" != "0" ]
+  then
+    ${KUBE_PV_CHOWN} mongo
+  fi
+
+  mkdir -p ${shard_dir}
+  chgrp 0 ${shard_dir}
+  chmod g=u ${shard_dir}
 
-  mkdir -p ${dir}
-  chgrp 0 ${dir}
-  chmod g=u ${dir}
 }
 
+prepare_postgres_pv() {
+  local dir="/var/lib/pgsql"
+  local uid=$(stat -c '%g' ${dir})  
+  local gid=$(stat -c '%u' ${dir})
+
+  # change ownership and permissions of mongo db path
+  if [ "${uid}" != "10001" ] || [ "${gid}" != "0" ]
```

Comment:
we are still not resilient to uid change, we just assume that the uid:gid will be 10001:0.
I think we should maybe pass the EXPECTED_UID and EXPECTED_GID as envs. this will give us more flexibility if a uid is changed